src/govuk/components/character-count/character-count.mjs

```diff
@@ -23,14 +24,49 @@ import { mergeConfigs, normaliseDataset } from '../../common.mjs'
  * @param {Number} [config.threshold=0] - The percentage value of the limit at
  * which point the count message is displayed. If this attribute is set, the
  * count message will be hidden by default.
+ * @param {Object} [config.i18n]
+ * @param {String} [config.i18n.charactersUnderLimitOne="You have %{count} character remaining"]
+ *   Message notifying users they're 1 character under the limit
+ * @param {String} [config.i18n.charactersUnderLimitOther="You have %{count} characters remaining"]
+ *   Message notifying users they're any number of characters under the limit
+ * @param {String} [config.i18n.charactersAtLimit="You have no characters remaining"]
+ *   Message notifying users they've reached the limit number of characters
+ * @param {String} [config.i18n.charactersOverLimitOne="You have %{count} character too many"]
+ *   Message notifying users they're 1 character over the limit
+ * @param {String} [config.i18n.charactersOverLimitOther="You have %{count} characters too many"]
+ *   Message notifying users they're any number of characters over the limit
+ * @param {String} [config.i18n.wordsUnderLimitOne="You have %{count} word remaining"]
+ *   Message notifying users they're 1 word under the limit
+ * @param {String} [config.i18n.wordsUnderLimitOther="You have %{count} words remaining"]
+ *   Message notifying users they're any number of words under the limit
+ * @param {String} [config.i18n.wordsAtLimit="You have no words remaining"]
+ *   Message notifying users they've reached the limit number of words
+ * @param {String} [config.i18n.wordsOverLimitOne="You have %{count} word too many"]
+ *   Message notifying users they're 1 word over the limit
+ * @param {String} [config.i18n.wordsOverLimitOther="You have %{count} words too many"]
```

Comment:
Thoughts on making large complicated JSDoc objects like **config.i18n** into a `@typedef`?

You can share them between files too, so uses of **config.i18n** match

Like the one we did for **ComponentData**?

```mjs
/**
 * Example
 *
 * @param {object} example
 * @param {import('../../lib/file-helper').ComponentData} example.componentData
 */
```

```mjs
/**
 * Component data from YAML
 *
 * @typedef {object} ComponentData
 * @property {string} name - Component name
 * @property {unknown[]} [params] - Nunjucks macro options
 * @property {unknown[]} [examples] - Example Nunjucks macro options
 * @property {string} [previewLayout] - Nunjucks layout for component preview
 * @property {string} [accessibilityCriteria] - Accessibility criteria
 */
```

Reply:
Just commenting to say we'll leave this (JSDoc `@typedef`) for now, but would be nice to see one coming soon

Reply:
They've been set up there: 
- https://github.com/alphagov/govuk-frontend/pull/2895/files#diff-cde937e9880b3f4528adf0fe09930dfdb170736d3cb092d42173c11c00995caaR1-R15
- https://github.com/alphagov/govuk-frontend/pull/2895/files#diff-604c43935cbbf4edb8c1209bc9b06ae7341fdaf64e208f777db723fd456bf9baR8-R16